crates/miden-tx/src/host/script_mast_forest_store.rs

```diff
@@ -38,6 +38,17 @@ impl ScriptMastForestStore {
             self.mast_forests.insert(proc_digest, mast_forest.clone());
         }
     }
+
+    /// Returns an iterator over unique MAST forests stored in this store.
+    pub fn forests(&self) -> impl Iterator<Item = &Arc<MastForest>> {
+        // Deduplicate forests since multiple procedure digests can point to the same forest
+        let mut seen = alloc::collections::BTreeSet::new();
+        self.mast_forests.values().filter(move |forest| {
+            // Use pointer address for deduplication since we're dealing with Arc<MastForest>
+            let ptr = Arc::as_ptr(forest) as usize;
+            seen.insert(ptr)
+        })
+    }
```

Comment:
Another option is to add something like `adv_data` field to the struct and build the advice map in the constructor. Basically, the struct could look like so:
```rust
pub struct ScriptMastForestStore {
    mast_forests: BTreeMap<Digest, Arc<MastForest>>,
    adv_data: AdviceMap,
}
```
This would result in some data duplication - but I'm not sure if that's a problem. Will make returning the advice data for all notes involved in a transaction much simpler though.